File: backend/internal/models/attendance.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Attendance struct {
	ID              string    `gorm:"primaryKey;type:varchar(36)" json:"id"`
	UserID          string    `gorm:"index;not null;type:varchar(36)" json:"user_id"`
	ClockIn         *time.Time `json:"clock_in"`
	ClockOut        *time.Time `json:"clock_out"`
	ClockInPhoto    string    `gorm:"type:varchar(500)" json:"clock_in_photo"`
	ClockOutPhoto   string    `gorm:"type:varchar(500)" json:"clock_out_photo"`
	ClockInLocation string    `gorm:"type:text" json:"clock_in_location"`
	ClockOutLocation string   `gorm:"type:text" json:"clock_out_location"`
	IsVerified      bool      `gorm:"default:false" json:"is_verified"`
	CreatedAt       time.Time `json:"created_at"`
	UpdatedAt       time.Time `json:"updated_at"`
	DeletedAt       gorm.DeletedAt `gorm:"index" json:"-"`
	
	User User `gorm:"foreignKey:UserID;constraint:OnDelete:CASCADE" json:"user,omitempty"`
}

File: lib/domain/entities/attendance.dart
import 'package:equatable/equatable.dart';

class Attendance extends Equatable {
  final String id;
  final String userId;
  final DateTime? clockIn;
  final DateTime? clockOut;
  final DateTime createdAt;
  final String? clockInPhoto;
  final String? clockOutPhoto;
  final String? clockInLocation;
  final String? clockOutLocation;
  final bool isVerified;
  
  const Attendance({
    required this.id,
    required this.userId,
    this.clockIn,
    this.clockOut,
    required this.createdAt,
    this.clockInPhoto,
    this.clockOutPhoto,
    this.clockInLocation,
    this.clockOutLocation,
    this.isVerified = false,
  });
  
  @override
  List<Object?> get props => [
    id,
    userId,
    clockIn,
    clockOut,
    clockInPhoto,
    clockOutPhoto,
    clockInLocation,
    clockOutLocation,
    isVerified,
  ];
}

